<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Annotations — annots_primary_hg38 • RLHub</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Annotations — annots_primary_hg38" />
<meta property="og:description" content="Genomic features relevant to R-loops. Both mm10 and
hg38 annotations are available." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">RLHub</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/RLHub.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/Bishop-Laboratory/RLHub/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Annotations</h1>
    <small class="dont-index">Source: <a href='https://github.com/Bishop-Laboratory/RLHub/blob/master/R/annotations.R'><code>R/annotations.R</code></a></small>
    <div class="hidden name"><code>annotations.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Genomic features relevant to R-loops. Both mm10 and
hg38 annotations are available.</p>
    </div>

    <pre class="usage"><span class='fu'>annots_primary_hg38</span><span class='op'>(</span>quiet <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>

<span class='fu'>annots_full_hg38</span><span class='op'>(</span>quiet <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>

<span class='fu'>annots_primary_mm10</span><span class='op'>(</span>quiet <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>

<span class='fu'>annots_full_mm10</span><span class='op'>(</span>quiet <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>quiet</th>
      <td><p>If TRUE, messages are suppressed. Default: FALSE.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A list of <code>tbl</code> objects. See details.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The <code>list</code> contains <code>tbl</code> objects (tidyverse-style data frames) containing
annotations as genomic ranges. The <strong>primary</strong> annotations (e.g.,
<code>annots_primary_hg38()</code>) are an abbreviated
version of the <strong>full</strong> annotations (e.g., <code>annots_full_hg38()</code>). See the
description below for further details:</p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Databases available</h3>


<p>This section details the annotation databases which are available in RLHub. See the succeeding
section ("Objects available based on accessor") for a list of which databases
are available within each function. All processing was performed using
<a href='https://github.com/Bishop-Laboratory/RLBase-data/blob/e63707e9e4dece5b2c007bd4284975be4998dfd4/scripts/getGenomicFeatures.R'>this script</a>
as part of the <a href='https://github.com/Bishop-Laboratory/RLBase-data'>RLBase-data</a> processing protocol.</p><ul>
<li><p><strong>Centromeres</strong></p><ul>
<li><p>Description: Centromere locations within the genome.</p></li>
<li><p>Source: UCSC table <a href='https://genome.ucsc.edu/cgi-bin/hgTables?db=hg38&amp;hgta_group=map&amp;hgta_track=centromeres&amp;hgta_table=centromeres&amp;hgta_doSchema=describe+table+schema'>centromeres</a>.</p></li>
</ul></li>
<li><p><strong>CNA</strong></p><ul>
<li><p>Description: Copy-number alterations found in inherited disorder cell lines. See source for full description. CNV states (0-4)
are represented in the data as separate types. For example, Deep deletion (0) sites are accessed with <code>annots_full_hg38()$CNA__0</code>.</p></li>
<li><p>Source:  UCSC table <a href='https://genome.ucsc.edu/cgi-bin/hgTables?db=hg38&amp;hgta_group=phenDis&amp;hgta_track=coriellDelDup&amp;hgta_table=coriellDelDup&amp;hgta_doSchema=describe+table+schema'>coriellDelDup</a>.</p></li>
</ul></li>
<li><p><strong>Cohesin</strong></p><ul>
<li><p>Description: This database contains manually-curated STAG2 and STAG1 ChIP-Seq data reprocessed by the RLHub authors to find
consensus STAG1 and STAG2 sites between cell lines (using <a href='https://github.com/rhysnewell/ChIP-R'>chip-r</a>.)</p></li>
<li><p>Source: processed data source <em><a href='https://academic.oup.com/nar/article/48/10/5639/5827199'>Pan et al</a></em> and processed
further as part of the RLBase-data pipeline <a href='https://github.com/Bishop-Laboratory/RLBase-data/tree/e63707e9e4dece5b2c007bd4284975be4998dfd4'>here</a>.</p></li>
</ul></li>
<li><p><strong>CpG_Islands</strong>:</p><ul>
<li><p>Description: CpG island predicted locations throughout the genome.</p></li>
<li><p>Source: UCSC table <a href='http://genome.ucsc.edu/cgi-bin/hgTables?hgta_doSchemaDb=hg38&amp;hgta_doSchemaTable=cpgIslandExt'>cpgIslandExt</a>.</p></li>
</ul></li>
<li><p><strong>Encode_CREs</strong>:</p><ul>
<li><p>Description: The UCSC Encode_CREs table contains putative promoter-like ("prom"), promoter-enhancer-like ("enhP"),
distal-enhancer-like ("enhD"), H3K4me3 ("K4me3"), and CTCF ("CTCF") chromatin states across the genome.</p></li>
<li><p>Source: UCSC table <a href='https://genome.ucsc.edu/cgi-bin/hgTables?db=hg38&amp;hgta_group=regulation&amp;hgta_track=encodeCcreCombined&amp;hgta_table=encodeCcreCombined&amp;hgta_doSchema=describe+table+schema'>encodeCcreCombined</a>.</p></li>
</ul></li>
<li><p><strong>Encode_Histone</strong>:</p><ul>
<li><p>Description: consensus peaks from histone ChIP experiments downloaded from Encode.
Biological replicates were summarized with <a href='https://github.com/rhysnewell/ChIP-R'>chip-r</a>.</p></li>
<li><p>Source: Encode v121. Manifest of samples downloaded is <a href='https://github.com/Bishop-Laboratory/RLBase-data/blob/e63707e9e4dece5b2c007bd4284975be4998dfd4/misc-data/histone_encode_manifest.csv'>here</a>.</p></li>
</ul></li>
<li><p><strong>encodeTFBS</strong>:</p><ul>
<li><p>Description: The collection of curated transcription-factor binding profiles from encode, made available by UCSC table browser.</p></li>
<li><p>Source: UCSC table <a href='https://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg38&amp;g=encRegTfbsClustered'>encRegTfbsClustered</a>.</p></li>
</ul></li>
<li><p><strong>G4Qexp</strong>:</p><ul>
<li><p>Description: G4-Quadruplex ChIP-Seq data</p></li>
<li><p>Source: GEO accession <a href='https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE63874'>GSE63874</a>.</p></li>
</ul></li>
<li><p><strong>G4Qpred</strong>:</p><ul>
<li><p>Description: Re-processed and binned G4-Quadruplex Predictions. The type names for this database are the G4Q prediction classes and follow the pattern <code>tl:N_nl:N_gn:N</code>.
<code>tl</code>: the length of guanine tracts in region; <code>nl</code>: number of locations for G4 formation; <code>gn</code>: the number of possible simultaneous G4 structures. For more information, see the source publication <a href='https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0165101'>here</a>.
Due to the large number of possible configurations of <code>tl</code>, <code>nl</code>, and <code>gn</code>, they were binned based on frequency.</p></li>
<li><p>Source: Figshare <a href='https://figshare.com/collections/G4_Supporting_Material/3498270/1'>Rouchka et al.</a> and <a href='https://figshare.com/ndownloader/files/6432597'>direct download link</a></p></li>
</ul></li>
<li><p><strong>knownGene_RNAs</strong>:</p><ul>
<li><p>Description: RNA species provided by UCSC KnownGene, split up by the "transcriptType" column from the source table.</p></li>
<li><p>Source: UCSC table <a href='http://genome.ucsc.edu/cgi-bin/hgTables?hgta_doSchemaDb=hg38&amp;hgta_doSchemaTable=knownGene'>knownGene</a>.</p></li>
</ul></li>
<li><p><strong>Microsatellite</strong>:</p><ul>
<li><p>Description: Microsatellite DNA regions predicted based on motif.</p></li>
<li><p>Source: UCSC table <a href='http://genome.ucsc.edu/cgi-bin/hgTables?hgta_doSchemaDb=hg38&amp;hgta_doSchemaTable=microsat'>microsat</a>.</p></li>
</ul></li>
<li><p><strong>PolyA</strong>:</p><ul>
<li><p>Description: List of predicted poly-A sites, split by the "name2" column of the source table.</p></li>
<li><p>Source: UCSC table <a href='http://genome.ucsc.edu/cgi-bin/hgTables?hgta_doSchemaDb=hg38&amp;hgta_doSchemaTable=wgEncodeGencodePolyaV38'>wgEncodeGencodePolyaV38</a>.</p></li>
</ul></li>
<li><p><strong>RBP_ChIP</strong>:</p><ul>
<li><p>Description: ChIP-Seq data sets for RNA-binding proteins (RBPs) generated by <a href='https://www.nature.com/articles/s41586-020-2077-3'><em>Nostrand et al</em></a> for Encode. Data are split by ChIP target.</p></li>
<li><p>Source: Encode v121. Manifest of samples <a href='https://static-content.springer.com/esm/art%3A10.1038%2Fs41586-020-2077-3/MediaObjects/41586_2020_2077_MOESM9_ESM.xlsx'>here</a> from source
study <a href='https://www.nature.com/articles/s41586-020-2077-3'><em>Nostrand et al</em></a>.</p></li>
</ul></li>
<li><p><strong>RBP_eCLiP</strong>:</p><ul>
<li><p>Description: eCLiP-Seq data sets for RNA-binding proteins (RBPs) generated by <a href='https://www.nature.com/articles/s41586-020-2077-3'><em>Nostrand et al</em></a> for Encode. Data are split by eCLiP target.</p></li>
<li><p>Source: Encode v121. Manifest of samples <a href='https://static-content.springer.com/esm/art%3A10.1038%2Fs41586-020-2077-3/MediaObjects/41586_2020_2077_MOESM6_ESM.xlsx'>here</a> from source
study <a href='https://www.nature.com/articles/s41586-020-2077-3'><em>Nostrand et al</em></a>.</p></li>
</ul></li>
<li><p><strong>Repeat_Masker</strong>:</p><ul>
<li><p>Description: Repeat masker table from UCSC containing genomic annotations for predicted repetitive elements, split by class of repetitive element ("repClass").</p></li>
<li><p>Source: UCSC table <a href='https://genome.ucsc.edu/cgi-bin/hgTables?db=hg38&amp;hgta_table=rmsk&amp;hgta_doSchema=describe+table+schema'>rmsk</a>.</p></li>
</ul></li>
<li><p><strong>skewr</strong>:</p><ul>
<li><p>Description: Regions of G or C-skew profiled using the <code>skewr</code> program. See the RLBase-data <a href='https://github.com/Bishop-Laboratory/RLBase-data'><code>README.md</code></a> for steps.</p></li>
<li><p>Source: From UCSC goldenPath, <a href='ftp://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz'>hg38</a> and
<a href='ftp://hgdownload.soe.ucsc.edu/goldenPath/mm10/bigZips/mm10.fa.gz'>mm10</a> genomes.
<a href='ftp://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/genes/hg38.ensGene.gtf.gz'>hg38</a>
and <a href='ftp://hgdownload.soe.ucsc.edu/goldenPath/mm10/bigZips/genes/mm10.ensGene.gtf.gz'>mm10</a> gene GTF.
CpG islands for mm10 and hg38 provided as described in the <strong>CpG_Islands</strong> entry above. Processing
proceeded using <a href='https://github.com/srhartono/SkewR/tree/e351e20c9d5a1704e7b5d79ad7dbdc895d04919d'><code>skewr</code></a> with
<a href='https://github.com/KorfLab/StochHMM/releases/tag/v0.38'><code>stochHMM v0.38</code></a>.</p></li>
</ul></li>
<li><p><strong>snoRNA_miRNA_scaRNA</strong>:</p><ul>
<li><p>Description: snoRNA, miRNA, and scaRNA species provided by UCSC table browser and split by the "type" column.</p></li>
<li><p>Source: UCSC table <a href='http://genome.ucsc.edu/cgi-bin/hgTables?hgta_doSchemaDb=hg38&amp;hgta_doSchemaTable=wgRna'>wgRna</a>.</p></li>
</ul></li>
<li><p><strong>Splice_Events</strong>:</p><ul>
<li><p>Description: UCSC table of alternative splice events predicted from transcriptome data sets. Split by "name" column.</p></li>
<li><p>Source: UCSC table <a href='http://genome.ucsc.edu/cgi-bin/hgTables?hgta_doSchemaDb=hg38&amp;hgta_doSchemaTable=knownAlt'>knownAlt</a>.</p></li>
</ul></li>
<li><p><strong>Transcript_Features</strong>:</p><ul>
<li><p>Description: Transcript features (e.g., "exon", "intron", etc) provided by Bioconductor TxDb packages. Split
based on the following features: "Exon", "Intron", "fiveUTR", "threeUTR", "TSS", "TTS", "Intergenic".</p></li>
<li><p>Source: TxDb for <a href='https://bioconductor.org/packages/release/data/annotation/html/TxDb.Hsapiens.UCSC.hg38.knownGene.html'>hg38</a> and
<a href='https://bioconductor.org/packages/release/data/annotation/html/TxDb.Mmusculus.UCSC.mm10.knownGene.html'>mm10</a>.</p></li>
</ul></li>
<li><p><strong>tRNAs</strong>:</p><ul>
<li><p>Description: UCSC table containing predicted tRNA genes.</p></li>
<li><p>Source: UCSC table <a href='http://genome.ucsc.edu/cgi-bin/hgTables?hgta_doSchemaDb=hg38&amp;hgta_doSchemaTable=tRNAs'>tRNAs</a>.</p></li>
</ul></li>
</ul>


<h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Objects available based on accessor function</h3>


<p>Here, we show which objects are available with each accessor function:</p><table class='table'>
<tr><td>DataBase name</td><td><code>annots_primary_hg38()</code></td><td><code>annots_primary_mm10()</code></td><td><code>annots_full_hg38()</code></td><td><code>annots_full_mm10()</code></td></tr>
<tr><td>Centromeres</td><td>FALSE</td><td>FALSE</td><td>TRUE</td><td>FALSE</td></tr>
<tr><td>CNA</td><td>FALSE</td><td>FALSE</td><td>TRUE</td><td>FALSE</td></tr>
<tr><td>Cohesin</td><td>FALSE</td><td>FALSE</td><td>TRUE</td><td>FALSE</td></tr>
<tr><td>CpG_Islands</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>
<tr><td>Encode_CREs</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>
<tr><td>Encode_Histone</td><td>FALSE</td><td>FALSE</td><td>TRUE</td><td>FALSE</td></tr>
<tr><td>encodeTFBS</td><td>FALSE</td><td>FALSE</td><td>TRUE</td><td>FALSE</td></tr>
<tr><td>G4Qexp</td><td>FALSE</td><td>FALSE</td><td>TRUE</td><td>FALSE</td></tr>
<tr><td>G4Qpred</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>FALSE</td></tr>
<tr><td>knownGene_RNAs</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>FALSE</td></tr>
<tr><td>Microsatellite</td><td>FALSE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td></tr>
<tr><td>PolyA</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>FALSE</td></tr>
<tr><td>RBP_ChIP</td><td>FALSE</td><td>FALSE</td><td>TRUE</td><td>FALSE</td></tr>
<tr><td>RBP_eCLiP</td><td>FALSE</td><td>FALSE</td><td>TRUE</td><td>FALSE</td></tr>
<tr><td>Repeat_Masker</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>
<tr><td>skewr</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>
<tr><td>snoRNA_miRNA_scaRNA</td><td>TRUE</td><td>FALSE</td><td>TRUE</td><td>FALSE</td></tr>
<tr><td>Splice_Events</td><td>FALSE</td><td>FALSE</td><td>TRUE</td><td>TRUE</td></tr>
<tr><td>Transcript_Features</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>
<tr><td>tRNAs</td><td>TRUE</td><td>TRUE</td><td>TRUE</td><td>TRUE</td></tr>
</table>




<h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Object structure</h3>


<p>Accessor functions (e.g., <code>annots_primary_hg38()</code>) return a named <code>list</code> of <code>tbl</code>
objects that specify feature ranges. Below, we detail the naming and structure of each.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>List names</h4>


<p>The names in the <code>list</code> objects provided by each accessor function (e.g., <code>annots_primary_hg38()</code>)
follow this structure: <code>DataBase__Type</code>. <code>DataBase</code> is the database from which
annotations were derived and <code>Type</code> indicates the specific annotations from the database
which are included in the <code>tbl</code>. This is required as some databases produce &gt; 1
type of annotation (e.g., <strong>Transcript_Features</strong> contains "Exon"
(<code>Transcript_Features__Exon</code>) and "Intron" (<code>Transcript_Features__Intron</code>)).</p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a><code>tbl</code> structure</h4>


<p>Each <code>tbl</code> returned has the following structure:</p><table class='table'>
<tr><td>chrom</td><td>start</td><td>end</td><td>strand</td><td>id</td></tr>
<tr><td>chr1</td><td>10015</td><td>10498</td><td>+</td><td>1</td></tr>
<tr><td>chr1</td><td>10614</td><td>11380</td><td>+</td><td>2</td></tr>
<tr><td>...</td><td></td><td></td><td></td><td></td></tr>
</table>



<p>Columns:</p><ul>
<li><p>"chom" - the Chromosome of the feature range (UCSC style)</p></li>
<li><p>"start" - the starting position of the feature range.</p></li>
<li><p>"end" - the end position of the feature range.</p></li>
<li><p>"strand" - the strand of the feature range.</p></li>
<li><p>"id" - A unique ID for the feature range.</p></li>
</ul>





    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='va'>annos</span> <span class='op'>&lt;-</span> <span class='fu'>annots_primary_hg38</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>snapshotDate(): 2021-10-05</span></div><div class='output co'>#&gt; <span class='message'>see ?RLHub and browseVignettes('RLHub') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='input'>
<span class='va'>annos</span> <span class='op'>&lt;-</span> <span class='fu'>annots_full_hg38</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>see ?RLHub and browseVignettes('RLHub') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='input'>
<span class='va'>annos</span> <span class='op'>&lt;-</span> <span class='fu'>annots_primary_mm10</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>see ?RLHub and browseVignettes('RLHub') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='input'>
<span class='va'>annos</span> <span class='op'>&lt;-</span> <span class='fu'>annots_full_mm10</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>see ?RLHub and browseVignettes('RLHub') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Henry Miller.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


