<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>RLBase Sample Manifest — rlbase_samples • RLHub</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="RLBase Sample Manifest — rlbase_samples" />
<meta property="og:description" content="A tbl containing metadata about each sample in RLBase." />
<meta property="og:image" content="/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">RLHub</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/RLHub.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/Bishop-Laboratory/RLHub/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>RLBase Sample Manifest</h1>
    <small class="dont-index">Source: <a href='https://github.com/Bishop-Laboratory/RLHub/blob/master/R/rlbase_samples.R'><code>R/rlbase_samples.R</code></a></small>
    <div class="hidden name"><code>rlbase_samples.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>A <code>tbl</code> containing metadata about each sample in RLBase.</p>
    </div>

    <pre class="usage"><span class='fu'>rlbase_samples</span><span class='op'>(</span>quiet <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>quiet</th>
      <td><p>If TRUE, messages are suppressed. Default: FALSE.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A <code>tbl</code>.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    
<h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Source</h3>


<p>RLBase samples were curated by hand in Excel from searching keywords such as "R-loops"
and "RNA:DNA hybrids" in GEO, SRA, and PubMed. Where R-loop mapping data
was publically available, entries were added to the Excel spreadsheet such that
every sample (SRX.../ERX.../GSM...) had it's own line. Information was noted for each
sample, such as the "mode" (the type of R-loop mapping it was) and the "Condition"
(e.g., "RNaseH1", "WKKD", etc). When genomic input controls were available, they
were manually matched to the experimental samples for which they could serve as a
background control during peak calling.</p>
<p>The up-to-date excel sheet is found <a href='https://github.com/Bishop-Laboratory/RLBase-data/blob/main/rlbase-data/rlbase_catalog.xlsx'>here</a>.</p>
<p>Throughout the process of analyzing the data (see <a href='https://github.com/Bishop-Laboratory/RLBase-data'>RLBase-data</a>), additional
metadata was added to the sample sheet (see structure for full account).</p>

<h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Structure</h3>


<p><code>rlbase_samples</code> is a <code>tbl</code> with the structure:</p><table class='table'>
<tr><td>rlsample</td><td>label</td><td>condition</td><td>mode</td><td>lab</td><td>tissue</td><td>genotype</td><td>other</td><td>PMID</td><td>group</td><td>family</td><td>ip_type</td><td>strand_specific</td><td>moeity</td><td>bisulfite_seq</td><td>file_type</td><td>experiment_original</td><td>control_original</td><td>study</td><td>name</td><td>paired_end</td><td>read_length</td><td>control</td><td>eff_genome_size</td><td>genome</td><td>prediction</td><td>discarded</td><td>numPeaks</td><td>expsamples</td><td>exp_matchCond</td><td>coverage_s3</td><td>peaks_s3</td><td>fastq_stats_s3</td><td>bam_stats_s3</td><td>report_html_s3</td><td>rlranges_rds_s3</td><td>rlfs_rda_s3</td></tr>
<tr><td>SRX1070676</td><td>POS</td><td>S96</td><td>DRIPc</td><td>Fred Chedin</td><td>NT2</td><td>WT</td><td>NT</td><td>27373332</td><td>rl</td><td>DRIP</td><td>S9.6</td><td>TRUE</td><td>RNA</td><td>FALSE</td><td>public</td><td>GSM1720613</td><td>NA</td><td>SRP059800</td><td>GSM1720613: NT2 DRIPc-seq, rep 1; Homo sapiens; OTHER</td><td>FALSE</td><td>50</td><td>NA</td><td>2706186140</td><td>hg38</td><td>POS</td><td>FALSE</td><td>34092</td><td>SRX1070685,SRX1070686</td><td>WT_NT_SRP059800_NT2</td><td>coverage/SRX1070676_hg38.bw</td><td>peaks/SRX1070676_hg38.broadPeak</td><td>fastq_stats/SRX1070676_hg38__fastq_stats.json</td><td>bam_stats/SRX1070676_hg38__bam_stats.txt</td><td>reports/SRX1070676_hg38.html</td><td>rlranges/SRX1070676_hg38.rds</td><td>rlfs_rda/SRX1070676_hg38.rlfs.rda</td></tr>
<tr><td>SRX1070677</td><td>POS</td><td>S96</td><td>DRIPc</td><td>Fred Chedin</td><td>NT2</td><td>WT</td><td>NT</td><td>27373332</td><td>rl</td><td>DRIP</td><td>S9.6</td><td>TRUE</td><td>RNA</td><td>FALSE</td><td>public</td><td>GSM1720614</td><td>NA</td><td>SRP059800</td><td>GSM1720614: NT2 DRIPc-seq, rep 2; Homo sapiens; OTHER</td><td>FALSE</td><td>50</td><td>NA</td><td>2706186140</td><td>hg38</td><td>POS</td><td>FALSE</td><td>22117</td><td>SRX1070685,SRX1070686</td><td>WT_NT_SRP059800_NT2</td><td>coverage/SRX1070677_hg38.bw</td><td>peaks/SRX1070677_hg38.broadPeak</td><td>fastq_stats/SRX1070677_hg38__fastq_stats.json</td><td>bam_stats/SRX1070677_hg38__bam_stats.txt</td><td>reports/SRX1070677_hg38.html</td><td>rlranges/SRX1070677_hg38.rds</td><td>rlfs_rda/SRX1070677_hg38.rlfs.rda</td></tr>
<tr><td>SRX1070678</td><td>POS</td><td>S96</td><td>DRIP</td><td>Fred Chedin</td><td>NT2</td><td>WT</td><td>NT</td><td>27373332</td><td>rl</td><td>DRIP</td><td>S9.6</td><td>FALSE</td><td>DNA</td><td>FALSE</td><td>public</td><td>GSM1720615</td><td>NA</td><td>SRP059800</td><td>GSM1720615: NT2 DRIP-seq, 1; Homo sapiens; OTHER</td><td>FALSE</td><td>50</td><td>NA</td><td>2706186140</td><td>hg38</td><td>POS</td><td>FALSE</td><td>73924</td><td>SRX1070685,SRX1070686</td><td>WT_NT_SRP059800_NT2</td><td>coverage/SRX1070678_hg38.bw</td><td>peaks/SRX1070678_hg38.broadPeak</td><td>fastq_stats/SRX1070678_hg38__fastq_stats.json</td><td>bam_stats/SRX1070678_hg38__bam_stats.txt</td><td>reports/SRX1070678_hg38.html</td><td>rlranges/SRX1070678_hg38.rds</td><td>rlfs_rda/SRX1070678_hg38.rlfs.rda</td></tr>
<tr><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td></tr>
</table>



<p>Column description:</p><ul>
<li><p><code>rlsample</code> - The unique ID of the sample, same as in the <a href='https://www.ncbi.nlm.nih.gov/sra'>SRA</a>.</p></li>
<li><p><code>label</code> - Label corresponding to the author-supplied condition of the sample. "POS" indicates the sample should robustly map R-loops, "NEG" indicates the opposite.</p></li>
<li><p><code>condition</code> - The specific condition for each sample.</p></li>
<li><p><code>mode</code> - The type of R-loop mapping for each sample.</p></li>
<li><p><code>lab</code> - The senior author on the publication from which the data was provided.</p></li>
<li><p><code>tissue</code> - The tissue condition for the sample.</p></li>
<li><p><code>genotype</code> - The sample's genotype.</p></li>
<li><p><code>other</code> - A column for other pertinent metadata provided by the authors.</p></li>
<li><p><code>PMID</code> - The PMID associated with the sample</p></li>
<li><p><code>group</code> - One of "rl" (R-loop mapping) or "exp" (Expression data/RNA-Seq).</p></li>
<li><p><code>family</code> - The family of the "mode" (e.g., "DRIP" includes "sDRIP", "DRIPc", "qDRIP", etc)</p></li>
<li><p><code>ip_type</code> - The IP type of the "mode" for each sample. One of "S9.6", "dRNH" (dead RNaseH1), or "None".</p></li>
<li><p><code>strand_specific</code> - Whether the sample is stranded.</p></li>
<li><p><code>moeity</code> - The moeity which was IP'd (if applicable)</p></li>
<li><p><code>bisulfite_seq</code> - Whether the data uses bisulfite conversion sequencing (e.g., "BisDRIP-Seq" samples)</p></li>
<li><p><code>file_type</code> - The type of data (always "public" for RLBase samples).</p></li>
<li><p><code>experiment_original</code> - The original name of this sample as entered by hand in the curated Excel spreadsheet (usually converted from GSM to SRX).</p></li>
<li><p><code>control_original</code> - Same as above for the accompanying control sample (if applicable.)</p></li>
<li><p><code>study</code> - The SRA study accession for this sample.</p></li>
<li><p><code>name</code> - The sample's name as entered in SRA.</p></li>
<li><p><code>paired_end</code> - A logical indicating whether the data is paired end.</p></li>
<li><p><code>read_length</code> - The read length for the sample.</p></li>
<li><p><code>control</code> - The RLBase ID of the genomic input control sample corresponding to this sample (if applicable)</p></li>
<li><p><code>eff_genome_size</code> - The effective genome size based on read length and genome (calculated with the <code>khmer</code> package) see relevant portion of RLBase-data protocol <a href='https://github.com/Bishop-Laboratory/RLBase-data#preliminary'>here</a>.</p></li>
<li><p><code>genome</code> - The UCSC genome ID for this sample.</p></li>
<li><p><code>prediction</code> - The prediction from running <code><a href='https://bishop-laboratory.github.io/RLSeq/reference/predictCondition.html'>RLSeq::predictCondition()</a></code>.</p></li>
<li><p><code>discarded</code> - A logical indicating whether this sample was discarded during model building for a mismatch with its "label" (see <a href='fft_model.html'>models</a>).</p></li>
<li><p><code>numPeaks</code> - The number of peaks called for this sample.</p></li>
<li><p><code>expsamples</code> - The IDs of any corresponding expression samples.</p></li>
<li><p><code>exp_matchCond</code> - The meta data used to match this sample to any corresponding expression samples (if applicable).</p><ul>
<li><p><strong>Method</strong>: Some R-loop mapping studies also had matched RNA-Seq data. In these cases,
they were also recorded with the same metadata (where applicable) as R-loop mapping samples. To match expression and R-loop samples,
the <code>study</code>, <code>tissue</code>, <code>genotype</code>, and <code>other</code> columns were compared iteratively for each R-loop sample. If all four
were a match with at least one expression samples, then those four columns would be assigned as the <code>exp_matchCond</code>. If only
three were available, then they would become the <code>exp_matchCond</code>. To see the order in which columns were checked
for possible matches, view the <code>buildExpression.R</code> script in the
<a href='https://github.com/Bishop-Laboratory/RLBase-data/blob/main/scripts/buildExpression.R#L25-L32'>RLBase-data repo</a>.
See also the section on <code>corr(R/PVal/PAdj)</code> column in <a href='rlregions_annot.html'>rlregions</a>.</p></li>
</ul></li>
<li><p><code>coverage_s3</code> - The location of the coverage tracks (<code>.bw</code>) in the AWS S3 bucket for RLBase data ('s3://rlbase_data/').</p></li>
<li><p><code>peaks_s3</code> - Same as above for peak files (<code>.broadPeak</code>)</p></li>
<li><p><code>fastq_stats_s3</code> - Same as above for fastq QC statistics data (<code>.json</code>).</p></li>
<li><p><code>bam_stats_s3</code> - Same as above for BAM QC statistics data (<code>.txt</code>).</p></li>
<li><p><code>report_html_s3</code> - Same as above for reports from <code><a href='https://bishop-laboratory.github.io/RLSeq/reference/report.html'>RLSeq::report()</a></code> (<code>.html</code>).</p></li>
<li><p><code>rlranges_rds_s3</code> - Same as above for RLRanges R objects, as in <code><a href='https://bishop-laboratory.github.io/RLSeq/reference/RLRanges.html'>RLSeq::RLRanges()</a></code> (<code>.rds</code>)</p></li>
<li><p><code>rlfs_rda_s3</code> - Same as above for <a href='rlfs_res.html'>rlfs_res</a> objects generated by <code><a href='https://bishop-laboratory.github.io/RLSeq/reference/analyzeRLFS.html'>RLSeq::analyzeRLFS()</a></code> (<code>.rda</code>).</p></li>
</ul>



    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples">
<span class='va'>rlsamples</span> <span class='op'>&lt;-</span> <span class='fu'>rlbase_samples</span><span class='op'>(</span><span class='op'>)</span>
</pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Henry Miller.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


