<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>R-loop Regions — rlregions_annot • RLHub</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="R-loop Regions — rlregions_annot" />
<meta property="og:description" content="R-loop regions (RL regions) are consensus sites of R-loop formation derived
from a meta-analysis of the R-loop mapping experiments in RLBase. RLHub includes
information about these regions (rlregions_meta), the overlap of genomic
features with RL regions (rlregions_annot), and the read count matrices
for each sample in RLBase within each RL region (rlregions_counts)." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">RLHub</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/RLHub.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/Bishop-Laboratory/RLHub/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>R-loop Regions</h1>
    <small class="dont-index">Source: <a href='https://github.com/Bishop-Laboratory/RLHub/blob/master/R/rlregions.R'><code>R/rlregions.R</code></a></small>
    <div class="hidden name"><code>rlregions.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>R-loop regions (RL regions) are consensus sites of R-loop formation derived
from a meta-analysis of the R-loop mapping experiments in RLBase. RLHub includes
information about these regions (rlregions_meta), the overlap of genomic
features with RL regions (rlregions_annot), and the read count matrices
for each sample in RLBase within each RL region (rlregions_counts).</p>
    </div>

    <pre class="usage"><span class='fu'>rlregions_annot</span><span class='op'>(</span>quiet <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>

<span class='fu'>rlregions_meta</span><span class='op'>(</span>quiet <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>

<span class='fu'>rlregions_counts</span><span class='op'>(</span>quiet <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>quiet</th>
      <td><p>If TRUE, messages are suppressed. Default: FALSE.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A list of <code>tbl</code> objects. See details.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>R-loop regions (RL regions) are consensus sites of R-loop formation derived
from a meta-analysis of the R-loop mapping experiments in RLBase. In <code>RLHub</code>,
we provide access to:</p><ul>
<li><p>rlregions_meta - The metadata describing these RL regions.</p></li>
<li><p>rlregions_annot - RL regions overlapped with genome features (see <a href='annots_primary_hg38.html'>annotations</a>).</p></li>
<li><p>rlregions_counts - RL region read counts across RLBase samples.</p></li>
</ul>

    <h2 class="hasAnchor" id="source"><a class="anchor" href="#source"></a>Source</h2>

    
<h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>RL Regions - rlregions_meta</h3>


<p>RL Regions were derived during <a href='https://github.com/Bishop-Laboratory/RLBase-data#get-r-loop-consensus'>this</a>
step of the RLBase-data <a href='https://github.com/Bishop-Laboratory/RLBase-data'>protocol</a>.</p>
<p>Here is a brief summary of the key processing steps.</p><ol>
<li><p>Human RLBase samples which were labeled as "POS", classified as "POS" (see <a href='rlfs_res.html'>rlfs_res</a>), and which
have at least 5000 peaks were selected.</p></li>
<li><p>Then, these peaks were then randomly downsampled to 5000 ranges each and aggregated.</p></li>
<li><p>Then, the hg38 genome was binned into 10bp bins and the number of aggregated peaks was
overlapped with each bin and counted, producing a bedGraph.</p></li>
<li><p>Then, macs3 was implemented to call peaks on the bedGraph file.</p></li>
<li><p>Finally, the resulting ranges were combined to create summary ranges.</p></li>
</ol>

<p><strong>Note</strong>: steps 2-4 were performed separately on datasets from dRNH (catalytically-dead RNaseH1) R-loop mapping experiments
and S9.6-based experiments due to the noted differences in R-loop mapping between these modalities. In step 5, they were combined to
create summary ranges, and the original source (dRNH or S9.6) was noted in the resulting metadata.</p>

<h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>RL Regions anno - rlregions_annot</h3>


<p>Once RL regions were generated, they were overlapped with the annotations found in <code><a href='annots_primary_hg38.html'>annots_full_hg38()</a></code>.
This was done separately for dRNH-derived, S9.6-derived, and combined RL regions.</p>

<h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>RL Regions Counts - rlregions_counts</h3>


<p>Once RL regions were generated, <code><a href='https://rdrr.io/pkg/Rsubread/man/featureCounts.html'>Rsubread::featureCounts()</a></code> was used to count
the number of overlapping reads from the <code>.bam</code> alignment files for each
sample in RLBase (<a href='rlbase_samples.html'>rlbase_samples</a>) within each RL region (only "combined" regions used).</p>
<p>These data were then VST-transformed and TPM was also calculated.</p>

    <h2 class="hasAnchor" id="structure"><a class="anchor" href="#structure"></a>Structure</h2>

    
<h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>RL Regions - rlregions_meta</h3>


<p><code>rlregions_meta()</code> is a <code>tbl</code> with the following structure:</p><table class='table'>
<tr><td>rlregion</td><td>location</td><td>is_rlfs</td><td>source</td><td>confidence_level</td><td>medSignalVal</td><td>medPVal</td><td>medQVal</td><td>avgSignalVal</td><td>avgPVal</td><td>avgQVal</td><td>numPeaks</td><td>nSamples</td><td>nStudies</td><td>nModes</td><td>nTissues</td><td>ip_types</td><td>nIPTypes</td><td>pct_case</td><td>avgNumPeaks</td><td>medNumPeaks</td><td>geneIDs</td><td>allGenes</td><td>mainGenes</td><td>is_repeat</td></tr>
<tr><td>All_RL64926</td><td>chrM:0-16569:.</td><td>TRUE</td><td>dRNH S96</td><td>53.21</td><td>1.962881</td><td>20.92420</td><td>17.01782</td><td>2.563848</td><td>653.89461</td><td>649.8001</td><td>52373</td><td>298</td><td>44</td><td>16</td><td>36</td><td>S9.6,dRNH,RNA-m6A,None</td><td>4</td><td>75.16779</td><td>80117.79</td><td>45063</td><td>NA</td><td>NA</td><td>NA</td><td>TRUE</td></tr>
<tr><td>All_RL19159</td><td>chr5:49656610-49661950:.</td><td>FALSE</td><td>dRNH S96</td><td>36.55</td><td>2.826610</td><td>19.16160</td><td>15.24025</td><td>3.282115</td><td>108.41481</td><td>104.6716</td><td>107393</td><td>261</td><td>43</td><td>15</td><td>34</td><td>S9.6,RNA-m6A,None,dRNH</td><td>4</td><td>77.01149</td><td>73058.96</td><td>43815</td><td>NA</td><td>NA</td><td>NA</td><td>TRUE</td></tr>
<tr><td>All_RL50251</td><td>chr16:46386300-46391280:.</td><td>FALSE</td><td>dRNH S96</td><td>31.68</td><td>3.258560</td><td>25.21965</td><td>21.00418</td><td>3.428755</td><td>87.33233</td><td>83.3639</td><td>133750</td><td>243</td><td>43</td><td>16</td><td>33</td><td>RNA-m6A,S9.6,dRNH,None</td><td>4</td><td>76.13169</td><td>65721.60</td><td>38878</td><td>NA</td><td>NA</td><td>NA</td><td>TRUE</td></tr>
<tr><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td></tr>
</table>



<p>Column description:</p><ul>
<li><p><code>rlregion</code> - The ID of the RL Region.</p></li>
<li><p><code>location</code> - The location of the RL Region. This column follows the pattern "chrom:start-stop:strand"</p></li>
<li><p><code>is_rlfs</code> - Logical indicating whether the RL region overlaps with an R-loop forming sequence.</p></li>
<li><p><code>source</code> - Indicates whether the RL region was discovered from catalytically-dead RNaseH1-based mapping (<code>dRNH</code>), S9.6-based mapping (<code>S96</code>), or both (<code>dRNH S96</code>).</p></li>
<li><p><code>confidence_level</code> - The percent of samples in which this range was found.</p></li>
<li><p><code>medSignalVal</code> - The median broadPeak "signalVal" (see <a href='https://genome.ucsc.edu/FAQ/FAQformat.html#format13'>UCSC specification</a>) of sample peaks in the RL region.</p></li>
<li><p><code>medPVal</code> - The median broadPeak -log10 pval (see <a href='https://genome.ucsc.edu/FAQ/FAQformat.html#format13'>UCSC specification</a>) of sample peaks in the RL region.</p></li>
<li><p><code>medQVal</code> - The median broadPeak -log10 qval (adjusted pval) (see <a href='https://genome.ucsc.edu/FAQ/FAQformat.html#format13'>UCSC specification</a>) of sample peaks in the RL region.</p></li>
<li><p><code>avgSignalVal/avgPVal/avgQVal</code> - same as above, but with mean instead of median.</p></li>
<li><p><code>medNumPeaks</code> - The median number of peaks for samples which overlapped with this RL Region.</p></li>
<li><p><code>avgNumPeaks</code> - Same as above with mean instead of median.</p></li>
<li><p><code>nSamples</code> - The number of samples which overlap with this peak.</p></li>
<li><p><code>nModes</code> - The number of different R-loop mapping methods in which this RL region was found.</p></li>
<li><p><code>nTissues</code> - The number of different tissues in which this RL region was found.</p></li>
<li><p><code>ip_types</code> - The "IP types" (S9.6, dRNH, or None) in which this RL Region was found.</p></li>
<li><p><code>nIPType</code> - The number of different "IP types" in which this RL region was found.</p></li>
<li><p><code>pct_case</code> - The percentage of samples which overlapped with this RL region that were predicted "POS" by <code><a href='https://rdrr.io/pkg/RLSeq/man/predictCondition.html'>RLSeq::predictCondition()</a></code>.</p></li>
<li><p><code>geneIDs</code> - The IDs of all genes overlapping with this RL Region.</p></li>
<li><p><code>allGenes</code> - The Gene Symbols of all genes overlapping with this RL Region.</p></li>
<li><p><code>mainGenes</code> - The Gene Symbols of all genes (which appear in pathway databases) overlapping with this RL Region.</p></li>
<li><p><code>is_repeat</code> - A logical indicating whether the RL Region occurs in a repetitive genomic region as determined by repeat masker (and also peri-centromeric regions were included.)</p></li>
</ul>


<h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>RL Regions anno - rlregions_annot</h3>


<p><code>rlregions_annot()</code> is a <code>tbl</code> with the following structure:</p><table class='table'>
<tr><td>rlregion</td><td>annotation</td></tr>
<tr><td>All_RL4615</td><td>CNA__3__4</td></tr>
<tr><td>All_RL4615</td><td>CNA__1__8</td></tr>
<tr><td>All_RL4615</td><td>G4Qexp__G4Q_PDS_NaK_GSE63874__95848</td></tr>
<tr><td>...</td><td>...</td></tr>
</table>



<p>Column description:</p><ul>
<li><p><code>rlregion</code> - The RL Region ID corresponding to the rlregions_meta table.</p></li>
<li><p><code>annotation</code> - The annotation which overlaps with the query RL region. Annotation
names follow the convention "DataBase__Type__ID" (see also <a href='annots_primary_hg38.html'>annotations</a>).</p></li>
</ul>


<h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>RL Regions Counts - rlregions_counts</h3>


<p><code>rlregions_counts()</code> is a <code>SummarizedExperiment</code> with the following structure:</p><ul>
<li><p><code>dim</code>: 64418 532</p></li>
<li><p><code>assays</code></p><ul>
<li><p><code>cts</code> - Raw read counts of each RLBase sample within each RL Region.</p></li>
<li><p><code>tpm</code> - The TPM normalization of <code>cts</code></p></li>
<li><p><code>vst</code> - The VST transform of <code>cts</code> (see also <code><a href='https://rdrr.io/pkg/DESeq2/man/vst.html'>DESeq2::vst()</a></code>).</p></li>
</ul></li>
<li><p><code>colData</code> - A <code>DataFrame</code> with structure:</p></li>
</ul><table class='table'>
<tr><td>experiment</td><td>label</td><td>strand_specific</td><td>paired_end</td><td>mode</td><td>prediction</td><td>discarded</td><td>numPeaks</td><td>bam</td><td>bam_avail</td></tr>
<tr><td>SRX113814</td><td>POS</td><td>FALSE</td><td>FALSE</td><td>DRIVE</td><td>NEG</td><td>TRUE</td><td>337</td><td>../RLBase-data/rlbase-data/rlpipes-out/bam/SRX113814/SRX113814_hg38.bam</td><td>TRUE</td></tr>
<tr><td>SRX113812</td><td>POS</td><td>FALSE</td><td>FALSE</td><td>DRIP</td><td>NEG</td><td>TRUE</td><td>78</td><td>../RLBase-data/rlbase-data/rlpipes-out/bam/SRX113812/SRX113812_hg38.bam</td><td>TRUE</td></tr>
<tr><td>SRX2675003</td><td>POS</td><td>TRUE</td><td>FALSE</td><td>R-ChIP</td><td>POS</td><td>FALSE</td><td>10331</td><td>../RLBase-data/rlbase-data/rlpipes-out/bam/SRX2675003/SRX2675003_hg38.bam</td><td>TRUE</td></tr>
<tr><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td></tr>
</table>



<p>Column description:</p><ul>
<li><p><code>experiment</code> - The RLBase sample ID</p></li>
<li><p><code>label</code> - The label provided by the data submitters ("POS" or "NEG")</p></li>
<li><p><code>strand_specific</code> - A logical indicating whether the data is stranded or not.</p></li>
<li><p><code>paired_end</code> - A logical indicating whether the data is paired-end or not.</p></li>
<li><p><code>prediction</code> - The predicted label from <code><a href='https://rdrr.io/pkg/RLSeq/man/predictCondition.html'>RLSeq::predictCondition()</a></code>. Can be "POS" or "NEG".</p></li>
<li><p><code>discarded</code> - A logical indicating whether the sample was discarded during model building due to a mismatch with its label.</p></li>
<li><p><code>numPeaks</code> - The number of peaks called from this sample.</p></li>
<li><p><code>bam</code> - The relative path of the bam file for this sample.</p></li>
<li><p><code>bam_avail</code> - A logical indicating whether the bam file was available.</p></li>
</ul>



    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='va'>rlregions</span> <span class='op'>&lt;-</span> <span class='fu'>rlregions_meta</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>see ?RLHub and browseVignettes('RLHub') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='input'>
<span class='va'>rlregionsAnno</span> <span class='op'>&lt;-</span> <span class='fu'>rlregions_annot</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>see ?RLHub and browseVignettes('RLHub') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='input'>
<span class='va'>rlregionsCounts</span> <span class='op'>&lt;-</span> <span class='fu'>rlregions_counts</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>see ?RLHub and browseVignettes('RLHub') for documentation</span></div><div class='output co'>#&gt; <span class='message'>loading from cache</span></div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Henry Miller.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


